### ISSUE TITLE
Some items are not internationalized

### ISSUE THREAD

**Issue by fw** [RPTools/maptool#2583]
**Is your feature request related to a problem? Please describe.**
Some texts are not internationalized. Many of them are dropdown list components, such as Token Size selection.

I couldn't determined category for this issue. Possibly, this may be fitted to place in bugs.

**Describe the solution you'd like**
To add missing i18n properties.

**Additional context**
Here are items that is not internationalized I have noticed.

#### Token Edit dialog
* Each items within Method selection at VBL Optimization in VBL Tab
* Each items within Shape, Size and Terrain Modifier selection and Ignored Terrain combo box in Config tab

#### Campaign Properties dialog
* Each items within Type and Corner selection in State tab
* Each items within Type and Side selection in Bar tab

#### Preferences
* Each items within New Map Grid Type and Light, Movement metric, Duplicate Token Numbering, Show Numbering on and New Token Naming in  the Interactions tab

#### Map Properties dialog
* Each items within AI Cell Rounding and Light

#### Token Context Menu
* Size categories (Medium, Small, and so on) within Size submenu on Token and Map Explorer

#### Draw Explorer
* All items in context menu of drawings

#### Drawing Tools Palette
* All tool-tips (such as Snap to grid, Line Type,  Eraser, Pen Width, Opacity)

#### Status Bar
* Tool-tips for Zoom Level and Used Memory

#### Map
* Following text labels: "Map not visible to players" and "Player View"

**Comment by Phergus** [comment]
Could be either but Feature is good enough.

**Comment by Irarara** [comment]
This is great. I've been keeping a list of things I've noticed as well and been wanting to tackle some of this stuff.

I suppose that makes this issue as good a place as any to ask a few questions I've had related to adding/editing i18n strings:
1. Is there a naming scheme that should be followed for keys or otherwise any preference about how things should be organized? I get the general gist of it but it's not always consistent. Specific example, the entire context menu for drawings needs to be added, but context menu items seem to be all over (token context menu items are`token.popup.menu.*`, init panel context menu items are `initPanel.*`, etc.) I can always just pick something to go with, but I just want to make sure I'm not going to make things disorganized.
    1. If you have something like say, a dialog window that's opened with a menu option, is it preferred to associate those things together (like, `foo.bar.dialog.*`) or organize them separately (both seem to be done?).
2. If I understand right, when the source strings are changed the strings for other languages are updated to reflect them. Is there a way to get around that if you're trying to change something that shouldn't affect the translations?
3. Likewise, is it possible to rename keys without creating a bunch of extra work for translators? (Not something I necessarily see a need to do, but just curious).

**Comment by Phergus** [comment]
1) No defined naming scheme for the i18n keys.  As you have seen different developers followed different schemes over the years. Some more alike than others.

2)  A change to a string does trigger the other translation files to get reset to the default for that key.  I am not aware of a way to avoid that happening.

3)  Haven't tried it but _maybe_ if the key changes were made to all i18 files and all were committed at the same time it wouldn't trigger the reset?

**Comment by Irarara** [comment]
So, similar to the case with layer names, for enums a common place where some things need working around is is in macro functions where some use the display name instead of the enum name (ie `getLayer()`, `getTokenShape()`, etc.) Should I make sure macros behave identically, or would it be acceptable to make some changes here?

The main reason I ask is because I kind of feel like it would be nice if there was some consistency about whether these sort of functions used the enum name or English display name (ie `TOP_DOWN` vs`Top Down`). For instance, `getLayer()` will return the enum names as is, but `getTokenShape()` returns the display names. (To note though, in the case of the token shape functions, the input for `setTokenShape()` is already case insensitive and will accept `top down` as input, so either form of output works as input).

Also, normally I'd probably just stick with whatever is on the wiki, but the documentation for `getTokenShape()` and `setTokenShape()` is actually incorrect at the moment.

**Comment by Phergus** [comment]
My thought is that the macro functions should be consistent in access these kind of things and that they should be using the enum name as any English string that is displayed should be localized.  As this wouldn't be just a simple move text to properties file and replace with I18n.getText() call, this work would be under a separate issue.  I wish GitHub had sub-issues.

**Comment by Irarara** [comment]
> My thought is that the macro functions should be consistent in access these kind of things and that they should be using the enum name as any English string that is displayed should be localized

That's my feeling as well, and so far most _do_ seem use the enum names for input and output.

I'll open an issue related to the macros sometime then. It'd be nice to do both at the same time because going out of the way to maintain the current behavior only to change it later seems kinda silly.

**Comment by Irarara** [comment]
In case I haven't asked enough questions... 
1. I feel like the corner option for states, where would sound more natural using top/bottom/left/right instead of cardinal directions as it is now (and that language is already used for bar side). Do I need to open an issue to propose a change to English text or would it be okay to go ahead and change that when localizing it?
2. Since this does require some tweaks beyond just swapping out strings, should I do this all in one PR or split it up? It's not going to be a huge amount of code or anything, but it will be touching a bunch of different files probably.

**Comment by Phergus** [comment]
1. Not sure I'm following on the corner states question.  North East is not any of top/bottom/left/right.  It would be Top Right, is that what you are thinking?

It wouldn't bother me to switch from compass directions but I do wonder if there are going to be layout issues after localizing the strings either way.  So many of the dialogs have fixed layout.

2. More work but breaking out the effort to localize areas like the States tab under a separate tickets would make it easier to track and manage.
3.

**Comment by Irarara** [comment]
>  It would be Top Right, is that what you are thinking?

Exactly. Sorry if that was vague, I meant using that wording as opposed to north/south/east/west. It'd be `Top Left`, `Top Right`, `Bottom Left`, `Bottom Right`. Layout in this particular case actually isn't an issue, the dropdown sizes to fit the largest item, and the dialog window seems to resize to fit the layout if it gets too big, it turns out (really: https://i.imgur.com/r8fJEAy.png). 

And alright, if it'd make it simpler to deal with I'll do that. I'm good either way. 👍

**Comment by Phergus** [comment]
Sounds good.  A lot of folks I know struggle with compass directions so this might help the directionally-challenged.  ;)

**Comment by Irarara** [comment]
So, there's a slight issue with localizing a few of the drop downs in the preferences dialog. With some of them (well, all of them except new map light and movement metric) the menu items are just strings, and the associated preference is saved just by storing that string (things like, `Token.NUM_ON_NAME`). The simplest solution seems like it would be to make enums (or an enum) to represent the options and use them to populate the drop downs. 

This would mean adjusting some things, but the main thing I want to know is how big of a deal it would be if I changed how the preferences are stored. If I switch over to representing the options as an enum, it feels like it would make more sense to store the enum name. but that would mean the user's current preferences for those wouldn't restore after the change (without something extra to handle the old values, of course, but at that point it feels like you almost may as well keep storing the old values?).

**Comment by Phergus** [comment]
@Irarara do you have more in the works for this?

**Comment by Irarara** [comment]
With that last PR this should be done, I believe I've gone through everything brought up here (and some). Let me know if anything got missed, but I think this covers all of the obvious stuff.

**Comment by Irarara** [comment]
@Phergus Tiny thing, but I realized I made a little mistake, the paint chooser dialog really should have its own unique title after all, since "paint" can be a color _or_ texture.  Would it be alright if I opened a PR to correct that?

I also spotted some unlocalized tooltips in preferences that got missed. I could throw those in too. (Of course I noticed them almost immediately after I thought I was done with everything. 🙃)
